src/osd/ECBackend.cc

```diff
@@ -1838,13 +1838,12 @@ bool ECBackend::try_state_to_reads()
     return false;
   }
 
+  op->using_cache = pipeline_state.caching_enabled();
+
```

Comment:
```c++
if (pipeline_state.caching_enabled()) {
  op->using_cache = true;
  if (op->invalidates_cache()) {
    dout(20) << __func__ << ": invalidating cache after this op"
             << dendl;
    pipeline_state.invalidate();
  }
}
```

@Aran85 thanks for this fix. it looks great! since you are at here, could you add another commit to avoid reset the state, if the cache is already invalidated?

Reply:
thanks the review! actually, the pipeline is more like be validated, so should we change the default value of ‘using_cache’ to true? it could avoid resetting the state further more.the write will not reenqueue the waiting_state, so I think it’s safe here. do you wind I chang the code in this commit and add your signed-off?

Reply:
> so should we change the default value of ‘using_cache’ to true?

we could. it saves a write access to the memory under most circumstances =)

> do you mind I chang the code in this commit and add your signed-off?

i'd suggest add another commit.